src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -3637,6 +3638,8 @@ def managed_instance_update(
     if tags is not None:
         instance.tags = tags
 
+    instance.maintenance_configuration_id = maintenance_configuration_id
```

Comment:
```suggestion
    if maintenance_configuration_id is not None:
        instance.maintenance_configuration_id = maintenance_configuration_id
```
We only need to update this when given value is not `None`, right?

Reply:
I think we don't need this check. If maintenance_configuration_id is not passed the API won't remove this property. We made sure it doesn't. The only way to remove property is to pass a sentinel value like this: `/subscriptions/a295933f-f7f5-4994-a109-8fa51241a5d6/providers/Microsoft.Maintenance/publicMaintenanceConfigurations/SQL_Default`

Reply:
Does this sound reasonable?